pkg/loader/compose/compose.go

```diff
@@ -272,12 +321,131 @@ func loadPorts(composePorts []string) ([]kobject.Ports, error) {
 	return ports, nil
 }
 
-// LoadFile loads compose file into KomposeObject
+// LoadFile loads a compose file into KomposeObject
 func (c *Compose) LoadFile(files []string) (kobject.KomposeObject, error) {
-	komposeObject := kobject.KomposeObject{
-		ServiceConfigs: make(map[string]kobject.ServiceConfig),
-		LoadedFrom:     "compose",
+
+	// Load the json / yaml file in order to get the version value
+	var version string
+
+	for _, file := range files {
+		composeVersion, err := getVersionFromFile(file)
```

Comment:
The checks have made code un-readable and kinda messy, so using the already existing loader mechanism should help I think, if we identify lot of repeating code we can abstract it for both type of loaders.